pkg/system/phase4_configuring.go

```diff
@@ -51,10 +57,110 @@ func (r *Reconciler) ReconcilePhaseConfiguring() error {
 	if err := r.ReconcileReadSystem(); err != nil {
 		return err
 	}
+	if err := r.ReconcileDeploymentEndpointStatus(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ReconcileDeploymentEndpointStatus creates/updates the endpoints deployment
+func (r *Reconciler) ReconcileDeploymentEndpointStatus() error {
+	if !util.KubeCheck(r.DeploymentEndpoint) {
+		return fmt.Errorf("Could not load endpoint deployment")
+	}
+	if r.DeploymentEndpoint.Status.ReadyReplicas == 0 {
+		return fmt.Errorf("First endpoint is not ready yet")
+	}
+
+	podSpec := &r.DeploymentEndpoint.Spec.Template.Spec
+	virtualHosts := []string{}
+	for i := range podSpec.Containers {
+		c := &podSpec.Containers[i]
+		if c.Name == "endpoint" {
+			for j := range c.Env {
+				e := c.Env[j]
+				if e.Name == "VIRTUAL_HOSTS" {
+					virtualHosts = append(virtualHosts, strings.Fields(e.Value)...)
+				}
+			}
+		}
+	}
+
+	r.NooBaa.Status.Endpoints = &nbv1.EndpointsStatus{
+		ReadyCount:   r.DeploymentEndpoint.Status.ReadyReplicas,
+		VirtualHosts: virtualHosts,
+	}
 
 	return nil
 }
 
+// SetDesiredDeploymentEndpoint updates the endpoint deployment as desired for reconciling
+func (r *Reconciler) SetDesiredDeploymentEndpoint() {
+	r.DeploymentEndpoint.Spec.Selector.MatchLabels["noobaa-s3"] = r.Request.Name
+	r.DeploymentEndpoint.Spec.Template.Labels["noobaa-s3"] = r.Request.Name
+
+	endpointsSpec := r.NooBaa.Spec.Endpoints
+	if endpointsSpec != nil {
+		replicas := int32(endpointsSpec.MinCount)
+		r.DeploymentEndpoint.Spec.Replicas = &replicas
+	}
+
+	podSpec := &r.DeploymentEndpoint.Spec.Template.Spec
+	for i := range podSpec.Containers {
+		c := &podSpec.Containers[i]
+		switch c.Name {
+		case "endpoint":
+			c.Image = r.NooBaa.Status.ActualImage
+			if endpointsSpec.Resources != nil {
+				c.Resources = *endpointsSpec.Resources
+			}
+
+			for j := range c.Env {
+				switch c.Env[j].Name {
+				case "MGMT_URL":
+					c.Env[j].Value = fmt.Sprintf(`https://%s.%s.svc`,
+						r.ServiceMgmt.Name, r.Request.Namespace)
+
+				case "MONGODB_URL":
+					// TODO: change to mongodb pod name after mongo separation
+					c.Env[j].Value = fmt.Sprintf(`mongodb://%s-0.%s/nbcore`,
+						r.CoreApp.Name, r.ServiceMgmt.Name)
```

Comment:
I think the pod dns name should be (r.NooBaaDB.Name)-0.(r.NooBaaDB.Spec.ServiceName)

Reply:
Changed the code as of your suggestion